### PR TITLE
script: Obtain milestones via github API

### PR DESCRIPTION
**Pull request by LenzGr**
Replace the hard-coded list of milestones with an API request that obtains this data from github directly.

Signed-off-by: Lenz Grimmer <lgrimmer@suse.com>

### CODE REVIEW

**Comment by smithfarm** [comment]
Thanks, @LenzGr ! LGTM but I won't be able to test it immediately. Give me some time.

File: src/script/ceph-backport.sh [hunk 1]
```diff
@@ -83,33 +83,13 @@ fi
 test "$milestone" || usage
 echo "Milestone: $milestone"
 
-# ------------------------------------
-# How to find out the milestone number
-# ------------------------------------
-# can't seem to extract the milestone number with the API
-# milestone numbers can be obtained with:
-#   curl --verbose -X GET https://api.github.com/repos/ceph/ceph/milestones
```
Comment (smithfarm) [inline src/script/ceph-backport.sh]: How about we leave part of this comment in?

Something like:

```
# milestone numbers can be obtained manually with:
#   curl --verbose -X GET https://api.github.com/repos/ceph/ceph/milestones
```

**Comment by tchaikov** [comment]
@LenzGr ping?

**Comment by smithfarm** [comment]
@LenzGr This PR adds code based on the following `curl` command:

```
curl -s GET https://api.github.com/repos/ceph/ceph/milestones
```

Up until the latest Leap 15.1 update, this code was working fine for me. However, it appears the latest update to curl broke it:

```
$ curl --version
curl 7.60.0 (x86_64-suse-linux-gnu) libcurl/7.60.0 OpenSSL/1.1.0i zlib/1.2.11 libidn2/2.0.4 libpsl/0.20.1 (+libidn2/2.0.4) libssh/0.8.4/openssl/zlib nghttp2/1.31.1
Release-Date: 2018-05-16
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS IDN IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz TLS-SRP HTTP2 UnixSockets HTTPS-proxy PSL Metalink 
$ curl -s GET https://api.github.com/repos/ceph/ceph/milestones | head -n1
The requested URL was not found on this server.[
(23) Failed writing body
```

Therefore, please prefix `-X` before `GET` in the `curl` commands you are adding:

```
$ curl -s -X GET https://api.github.com/repos/ceph/ceph/milestones | head -n1
[
```

**Comment by LenzGr** [comment]
>     * please preserve two lines of documentation for posterity
>     * please prefix `GET` with `-X` in the curl commands being added

Will do, thanks for the feedback and sorry for the delay.

**Comment by LenzGr** [comment]
>     * please preserve two lines of documentation for posterity
>     * please prefix `GET` with `-X` in the curl commands being added

Done, please re-review/approve!

File: src/script/ceph-backport.sh [hunk 2]
```diff
@@ -90,33 +90,16 @@ fi
...
-    milestone_number=12
-    target_branch=nautilus
+milestone_number=$(curl -s -X GET https://api.github.com/repos/ceph/ceph/milestones | jq --arg milestone $milestone '.[] | select(.title==$milestone) | .number')
+
+if test -n "$milestone_number" ; then
+    target_branch="$milestone"
 else
-    echo "Please enter hammer, jewel, kraken, luminous, mimic, or nautilus"
+    echo -n "Unknown Milestone. Please use one of the following ones: "
+    echo $(curl -s GET https://api.github.com/repos/ceph/ceph/milestones | jq '.[].title')
```
Comment (smithfarm) [inline src/script/ceph-backport.sh]: need `-X` here, too

Comment (LenzGr) [reply src/script/ceph-backport.sh]: Gah! Thanks, will update.

Comment (LenzGr) [reply src/script/ceph-backport.sh]: Fixed.

**Comment by smithfarm** [comment]
jenkins test docs

**Pull request merged**